packages/theme-check-common/src/checks/valid-block-preset-settings/index.ts

```diff
@@ -0,0 +1,115 @@
+import { isSection, isBlock } from '../../to-schema';
+import { basename } from '../../path';
+import { LiquidCheckDefinition, Severity, SourceCodeType, ThemeBlock } from '../../types';
+import { Preset } from '../../types/schemas/preset';
+
+export const ValidBlockPresetSettings: LiquidCheckDefinition = {
+  meta: {
+    code: 'ValidBlockPresetSettings',
+    name: 'Reports invalid preset settings for a theme block',
+    docs: {
+      description: 'Reports invalid preset settings for a theme block',
```

Comment:
Is it only theme blocks or just blocks in general?

Reply:
Or, whatever block is in the presets settings.